### PR TITLE
Added unit test for the default and the "increased" header size

### PR DESCRIPTION
**Pull request by paulrutter**
See https://github.com/creationix/http-parser-js/pull/29.
I've added two unit tests; one for the default maxHeaderSize (which should fail), and one test for the increased header size.

### CODE REVIEW

File: tests/parallel/test-http-client-max-header-size_default.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+server.listen(0, function() {
+  var options = {
+    port: server.address().port,
+    method: 'GET',
+    headers: header
+  };
+
+  var r = http.request(options);
+  r.on('error', function(err) {
```
Comment (Jimbly) [inline tests/parallel/test-http-client-max-header-size_default.js]: If no error occurs, this test will still pass!  I think you want to add something like:
```javascript
var had_error;
r.on('error', function(err) { had_error = true; });
process.on('exit', function() { assert(had_error); });
```

**Comment by paulrutter** [comment]
Ah you're right, thats not ok.
Will fix tomorrow.

**Comment by Jimbly** [comment]
Otherwise, looks good, thanks for adding a test!

**Comment by paulrutter** [comment]
Should be fixed now.

**Pull request merged**